src/app/GlobalAttributes.cpp

```diff
@@ -0,0 +1,132 @@
+        if (err != CHIP_NO_ERROR)
+        {
+            break;
+        }
+        auto buffer = builder.TakeBuffer();
+
+        return encoder.EncodeList([&buffer](const auto & listEncodeHelper) {
+            for (auto id : buffer)
+            {
+                ReturnErrorOnFailure(listEncodeHelper.Encode(id));
+            }
+            return CHIP_NO_ERROR;
+        });
+    }
+    case Clusters::Globals::Attributes::AcceptedCommandList::Id: {
+        DataModel::ListBuilder<DataModel::AcceptedCommandEntry> builder;
+        err = provider->AcceptedCommands(path, builder);
+        if (err != CHIP_NO_ERROR)
+        {
+            break;
+        }
+        auto buffer = builder.TakeBuffer();
+
+        return encoder.EncodeList([&buffer](const auto & listEncodeHelper) {
+            for (auto entry : buffer)
+            {
+                ReturnErrorOnFailure(listEncodeHelper.Encode(entry.commandId));
+            }
+            return CHIP_NO_ERROR;
+        });
+    }
+    case Clusters::Globals::Attributes::AttributeList::Id: {
+        DataModel::ListBuilder<DataModel::AttributeEntry> builder;
+        err = provider->Attributes(path, builder);
+        if (err != CHIP_NO_ERROR)
+        {
+            break;
+        }
+        auto buffer = builder.TakeBuffer();
+
+        return encoder.EncodeList([&buffer](const auto & listEncodeHelper) {
+            bool addedExtraGlobals      = false;
+            constexpr auto lastGlobalId = GlobalAttributesNotInMetadata[ArraySize(GlobalAttributesNotInMetadata) - 1];
+            // If EventList is not supported. The GlobalAttributesNotInMetadata is missing one id here.
+            static_assert(lastGlobalId - GlobalAttributesNotInMetadata[0] == ArraySize(GlobalAttributesNotInMetadata),
+                          "Ids in GlobalAttributesNotInMetadata not consecutive (except EventList)");
+
+            for (auto entry : buffer)
+            {
+                if (!addedExtraGlobals && entry.attributeId > lastGlobalId)
+                {
+                    for (const auto & globalId : GlobalAttributesNotInMetadata)
+                    {
+                        ReturnErrorOnFailure(listEncodeHelper.Encode(globalId));
+                    }
+                    addedExtraGlobals = true;
+                }
+                ReturnErrorOnFailure(listEncodeHelper.Encode(entry.attributeId));
+            }
+
+            if (!addedExtraGlobals)
+            {
+                for (auto id : GlobalAttributesNotInMetadata)
+                {
+                    ReturnErrorOnFailure(listEncodeHelper.Encode(id));
+                }
+            }
+            return CHIP_NO_ERROR;
+        });
+    }
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // if we get here, the path was NOT valid
+    if (err == CHIP_ERROR_NOT_FOUND)
+    {
+        return DataModel::ValidateClusterPath(provider, path, Status::Failure);
```

Comment:
Worth commenting that the Status::Failure is arbitrary, because in practice we clearly failed on the endpoint-or-cluster level above?